### PR TITLE
Api module + expanded route support

### PR DESCRIPTION
**Pull request by wpears**
Adds on more routing goodies to #99.

Also adds a rudimentary module which shall be the gateway to the API (to further isolate changes which will crop up here).

Will support soon-to-arrive submission and edit component changes

### CODE REVIEW

File: src/js/AppContainer.jsx [hunk 1]
```diff
@@ -6,19 +6,20 @@ var AppContainer = React.createClass({
...
+    var params = this.props.params || {};
 
     return (
       <div className="AppContainer">
-        <HomeLink/>
-        <UserHeading institution={{name: params.encodedInstitutionName}} year="2017" user={this.state.name}/>
-        {this.props.children}
+        <HomeLink year={params.year}/>
+        <UserHeading year={params.year} user={this.state.name}/>
+        {React.cloneElement(this.props.children, this.state.institution?{institution:this.state.institution}:null)}
```
Comment (awolfe76) [inline src/js/AppContainer.jsx:22]: Just want to be sure here, so rather than having all `props` this will now only give us the `institution`?

**Comment by awolfe76** [comment]
Other than my one question should be good.

File: src/js/AppContainer.jsx (hunk 1 above)
Comment (wpears) [inline src/js/AppContainer.jsx:22]: Actually before no props were being passed, now the institution is when it is present

Comment (awolfe76) [inline src/js/AppContainer.jsx:22]: 👍

**Pull request merged**